File: bench/bench.py
import time

# Fibonacci
def fib(n):
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)

start = time.time()
result = fib(30)
elapsed = time.time() - start
print(f"Fibonacci(30) = {result}")
print(f"Time: {elapsed:.3f}s")

# Loop
iterations = 1000000
start = time.time()
sum_val = 0
for i in range(iterations):
    sum_val += i
elapsed = time.time() - start
print(f"\nLoop sum: {sum_val}")
print(f"Iterations: {iterations}")
print(f"Time: {elapsed:.3f}s")

# Array operations
size = 10000
start = time.time()

arr = list(range(size))

arr_sum = sum(arr)

# Sort
sorted_arr = sorted(arr)

elapsed = time.time() - start
print(f"\nArray size: {size}")
print(f"Sum: {arr_sum}")
print(f"Sorted[0], Sorted[last]: {sorted_arr[0]}, {sorted_arr[-1]}")
print(f"Time: {elapsed:.3f}s")

# Function calls
iterations = 500000

def add(a, b):
    return a + b

start = time.time()
result = 0
for i in range(iterations):
    result = add(i, i + 1)
elapsed = time.time() - start
print(f"\nFunction calls: {iterations}")
print(f"Result: {result}")
print(f"Time: {elapsed:.3f}s")

File: bench/bench.go
package main

import (
	"fmt"
	"time"
)

func fib(n int) int {
	if n <= 1 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	// Fibonacci
	start := time.Now()
	result := fib(30)
	elapsed := time.Since(start)
	fmt.Printf("Fibonacci(30) = %d\n", result)
	fmt.Printf("Time: %v\n", elapsed)

	// Loop
	iterations := 1000000
	start = time.Now()
	sum := 0
	for i := 0; i < iterations; i++ {
		sum += i
	}
	elapsed = time.Since(start)
	fmt.Printf("\nLoop sum: %d\n", sum)
	fmt.Printf("Iterations: %d\n", iterations)
	fmt.Printf("Time: %v	// Array operations\n", elapsed)

	size := 10000
	start = time.Now()

	arr := make([]int, size)
	for i := 0; i < size; i++ {
		arr[i] = i
	}

	arrSum := 0
	for _, v := range arr {
		arrSum += v
	}

	// Simple bubble sort
	sorted := make([]int, len(arr))
	copy(sorted, arr)
	n := len(sorted)
	for i := 0; i < n; i++ {
		for j := 0; j < n-i-1; j++ {
			if sorted[j] > sorted[j+1] {
				sorted[j], sorted[j+1] = sorted[j+1], sorted[j]
			}
		}
	}

	elapsed = time.Since(start)
	fmt.Printf("\nArray size: %d\n", size)
	fmt.Printf("Sum: %d\n", arrSum)
	fmt.Printf("Sorted[0], Sorted[last]: %d, %d\n", sorted[0], sorted[len(sorted)-1])
	fmt.Printf("Time: %v\n", elapsed)

	// Function calls
	iterations = 500000
	start = time.Now()

	add := func(a, b int) int {
		return a + b
	}

	result = 0
	for i := 0; i < iterations; i++ {
		result = add(i, i+1)
	}
	elapsed = time.Since(start)
	fmt.Printf("\nFunction calls: %d\n", iterations)
	fmt.Printf("Result: %d\n", result)
	fmt.Printf("Time: %v\n", elapsed)
}
